src/point_one/fusion_engine/common/portability.h

```diff
@@ -35,8 +35,27 @@
 #  endif
 #endif
 
-// ssize_t is a POSIX extension and is not supported on Windows.
-#if defined(_WIN32)
+// Support ARM CC quirks
+#ifdef __CC_ARM
+#  include "typedefs.h"
+// ARM CC bug http://www.keil.com/forum/60227/
+#  define __ESCAPE__(__x) (__x)
+#  define P1_HAVE_STD_FUNCTION 0
+#  define P1_HAVE_STD_OSTREAM 0
+#  define P1_HAVE_STD_SMART_PTR 0
+#  define P1_NO_LOGGING 1
+#endif
+
+// Different compilers support different ways of specifying default struct
+// alignment. Since there's no universal method, a macro is used instead.
+#ifdef __CC_ARM
+#  define P1_ALIGNAS(N) __attribute__((aligned(N)))
+#else
+#  define P1_ALIGNAS(N) alignas(N)
```

Comment:
Just curious, but regular GCC supports `__attribute__((aligned(N)))` in addition to `alignas(N)`. Is there any benefit to using `alignas`?

Reply:
We originally switched to `alignas` since it worked for MVCC, Clang, and GCC. I think it was MVCC that doesn't support `__attribute__((aligned(N)))`. I figure leaving the "most popular" version as the default makes sense.

Reply:
At some point we should byte the bullet and define a macro for "structs that have correct padding to align their members, but use 4 byte alignment overall, and don't add implicit padding". If we're using a custom macro already it's not that much more complicated to have:

```
P1_START_PACKED_STRUCT Foo{

} P1_END_PACKED_STRUCT
```

or something similar to actually do this right and get compile time errors if you mess it up.